### PR TITLE
add software requirements specification

### PR DESCRIPTION
**Pull request by reikookamoto**
Draft of SRS in response to feedback from @yulric 

- Removed things not directly related to the goal of the function
- Added all arguments that the function currently accepts
- Switched order of the Feature and Behaviour sections
- Revised the Feature section to follow the BDD framework, and broke down interal-related features into sub-sections (closed, open, left-open right-closed, left-closed right-open)

### CODE REVIEW

**Review by yulric — COMMENTED** [review]
1. Is there a reason everything is in points? 
2. For the examples, we should try to keep them as small as possible while still being able to communicate what the feature is about. I can appreciate the need to start with concrete examples but for the final SRS I think they should be shortened so they're easier to read.

File: srs/srs.md [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+    4 DHHGAGE_5      4  [12,15]
+    5 DHHGAGE_5      5       16
+    6 DHHGAGE_5  NA::a       96
+    7 DHHGAGE_5  NA::b  [97,99]
+    8 DHHGAGE_5  NA::b     else
+
+Sample data (before re-coding):
+
+      DHHGAGE
+    1      16
```
[the remark this answers is not in the document]
Comment (reikookamoto) [reply srs/srs.md]: Done

**Comment by reikookamoto** [comment]
@yulric 

> 1. Is there a reason everything is in points?

- Thought it would make the document easier to read
- The BDD Testing article you shared also used bullet points to write out the Given, When and Then of each scenario

> 2. For the examples, we should try to keep them as small as possible while still being able to communicate what the feature is about. I can appreciate the need to start with concrete examples but for the final SRS I think they should be shortened so they're easier to read.

- Another option would be to render the document to HTML so we can leverage more Quarto features. Using [foldable callouts](https://quarto.org/docs/authoring/callouts.html) would allow us to keep the concrete examples while giving the reader the option to focus on one at a time.

**Comment by yulric** [comment]
Closing for now, [scoping-doc](https://github.com/Big-Life-Lab/recodeflow/tree/scoping-doc) is a continuation of this feature.

**Pull request closed without merging**